### PR TITLE
Quick start :)

### PR DESCRIPTION
**Pull request by SubhadityaMukherjee**
Hello! 
I made a little quickstart guide to Documenter.jl  (Previously mentioned in the [forums](https://discourse.julialang.org/t/step-by-step-documentation-tutorial/46093/2)).

I am not particularly sure where you need the file placed but I added it as a quickstart.md file. 
Hope it helps out people who are starting out with Julia and this awesome package :smile:

### CODE REVIEW

File: docs/src/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    - Follow this structure if you can (most of this is auto generated)
+      - DDL.jl
+        - src
+          - DDL.jl
+          - classif.jl
+        - docs
+        - tests
+        - README.md
+        - Manifest.toml
+        - Project.toml
```
Comment (MichaelHatherly) [inline docs/src/quickstart.md:38]: Put this directory structure in a code block rather, since it's nested in a list already the additional nesting doesn't make it stand out enough.

File: docs/src/index.md [hunk 2]
```diff
@@ -25,10 +25,13 @@ A package for building documentation from docstrings and markdown files.
 
 The [Package Guide](@ref) provides a tutorial explaining how to get started using Documenter.
 
+A quickstart version of the guide can be found at [Quickstart]("https://github.com/JuliaDocs/Documenter.jl/blob/master/docs/src/quickstart.md")
+
 Some examples of packages using Documenter can be found on the [Examples](@ref) page.
 
 See the [Index](@ref main-index) for the complete list of documented functions and types.
 
+
```
Comment (MichaelHatherly) [inline docs/src/index.md:34]: Unintentional extra blank?

**Review by MichaelHatherly — CHANGES_REQUESTED** [review]
Thanks @SubhadityaMukherjee. I've added a few inline comments.

I can understand the need for more tutorial/quickstart material for getting started with this kind of stuff, especially more "conversational" content. How do you feel it compares to the content of https://juliadocs.github.io/Documenter.jl/stable/man/guide/? What did you find was missing from that, might be worth iterating of it rather than increasing the size of the docs further?

File: docs/src/index.md [hunk 3]
```diff
@@ -25,10 +25,13 @@ A package for building documentation from docstrings and markdown files.
 
 The [Package Guide](@ref) provides a tutorial explaining how to get started using Documenter.
 
+A quickstart version of the guide can be found at [Quickstart]("https://github.com/JuliaDocs/Documenter.jl/blob/master/docs/src/quickstart.md")
```
[the remark this answers is not in the document]
Comment (SubhadityaMukherjee) [reply docs/src/index.md:28]: Hello @MichaelHatherly . Thank you for the review. :)
I felt that as documentation for someone who knows their way around these things it was perfect. But for someone just starting out with Julia/actually developing, it would be a bit hard to follow as many steps that were not related to Documenter.jl were involved. 
I had initially written a post on my blog but I was [told](https://discourse.julialang.org/t/step-by-step-documentation-tutorial/46093) that it might be a good addition to the documentation here so the PR.

I will fix whatever you mentioned by tomorrow. 
Cheers

**Comment by MichaelHatherly** [comment]
Quoting @SubhadityaMukherjee's reply outside of inline comments:

> Hello @MichaelHatherly . Thank you for the review. :)
> I felt that as documentation for someone who knows their way around these things it was perfect. But for someone just starting out with Julia/actually developing, it would be a bit hard to follow as many steps that were not related to Documenter.jl were involved.
> I had initially written a post on my blog but I was [told](https://discourse.julialang.org/t/step-by-step-documentation-tutorial/46093) that it might be a good addition to the documentation here so the PR.
> 
> I will fix whatever you mentioned by tomorrow.
> Cheers

Yes, I followed along in that thread and agree with @mortenpi's final comments.

Let's decide exactly *where* this content fits into the rest of the manual so that it maintains an overall "flow" rather than bifurcating the manual and resulting in duplication. I'd say it probably fits in right at the start of the "Guide" section and shouldn't be very long so that it actually results in a quick start to using the package. Using as much automation in it as well, `DocumenterTools` or `PkgTemplates`, should be required and not an optional "skip these steps" kind of thing, so that the steps are kept to a minimum. If we've got the tools to set things up automatically we shouldn't be instructing users to create specific files manually.

> I will fix whatever you mentioned by tomorrow.

Let's also get @mortenpi's take on this prior to doing any fixes, just so you don't have to redo things too much.

**Comment by mortenpi** [comment]
Apologies that it took so long for me to reply to this. I do agree with @MichaelHatherly's comments that we should try to see how we can improve the current "Guide" page. I also think it's worth keeping the [Divio 4-tiered documentation approach](https://documentation.divio.com/introduction/) in mind when figuring this out.

> I'd say it probably fits in right at the start of the "Guide" section and shouldn't be very long so that it actually results in a quick start to using the package. Using as much automation in it as well, `DocumenterTools` or `PkgTemplates`, should be required and not an optional "skip these steps" kind of thing, so that the steps are kept to a minimum. If we've got the tools to set things up automatically we shouldn't be instructing users to create specific files manually.

Yes, I absolutely agree with this.

1. I think the quickstart should really emphasize PkgTemplates. It does the right thing and removes a bunch of fiddly steps for the user.
2. It might be good to briefly show the user how to use project environments and explain that `docs/` is a separate environment, because I think a lot of users are not familiar and/or comfortable with them.
3. I do feel that the Guide page is a bit long. Some of the topics there could probably be expanded upon and split into separate pages, with references from the guide/quickstart page.

However, we shouldn't make this PR too big. Perhaps we could condense this particular PR down to be a PkgTemplates-based quickstart at the start of the Guide page? And then make additional improvements in subsequent PRs?

File: docs/src/quickstart.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+5. Now in the docs folder make this heirarchy
```
Comment (haampie) [inline docs/src/quickstart.md:39]:
```suggestion
5. Now in the docs folder make this hierarchy
```

**Comment by knuesel** [comment]
Would be nice to merge this on top of #1451! @SubhadityaMukherjee are you still intrested in working on this?

**Comment by mortenpi** [comment]
Let's close this PR, but hopefully we can address the points raised here as part of #2321.
